src/confluent_kafka/src/Consumer.c

```diff
@@ -1356,17 +1479,67 @@ static void Consumer_rebalance_cb (rd_kafka_t *rk, rd_kafka_resp_err_t err,
 		}
 	}
 
+        if (!strcmp(rd_kafka_rebalance_protocol(rk), "COOPERATIVE")) {
+                if (self->u.Consumer.rebalance_assigned) {
+                        assign_errstr = "assign() must not be called during a "
+                                        "COOPERATIVE protocol rebalance";
+                } else if (self->u.Consumer.rebalance_incremental_unassigned &&
+                           err == RD_KAFKA_RESP_ERR__ASSIGN_PARTITIONS) {
+                        assign_errstr = "incremental_unassign() must not be "
+                                        "called during an ASSIGN rebalance ";
+                } else if (self->u.Consumer.rebalance_incremental_assigned &&
+                           err == RD_KAFKA_RESP_ERR__REVOKE_PARTITIONS) {
+                        assign_errstr = "incremental_assign() must not be "
+                                        "called during a REVOKE rebalance";
+                }
+        } else {
+                if (self->u.Consumer.rebalance_incremental_assigned ||
+                    self->u.Consumer.rebalance_incremental_unassigned) {
+                        assign_errstr = "incremental_(un)assign must not be "
+                                        "called during an EAGER protocol "
+                                        "rebalance";
+                }
+        }
+
+        if (assign_errstr) {
+                cfl_PyErr_Format(RD_KAFKA_RESP_ERR__FAIL, assign_errstr);
```

Comment:
Whytespaces, not tabbs

Reply:
Must pass format string to format functions.

Reply:
otoh this code is not needed at all, see previous comment about enforcement in the librd of kafka

Reply:
yep, it's all enforced in librdkafka. thanks for pointing that out ...